### DISCUSSION TITLE
微信小程序 防抖函数如何获取传递的参数？

### DISCUSSION THREAD

**Discussion by KouSum** [remaxjs/remax#1420] [Q&A]
**问题描述**
微信小程序 防抖函数无法获取传递的参数

**示例代码**
`
export default function (props) {

    function debounce (fn, delay) {
        let timer ;
        return function(){
            const context = this
            if (timer) {
                clearTimeout(timer)
            }
            timer = setTimeout(function() {
                fn.apply(context,arguments)
            }, delay)
        }
    }
    function inputChange(e) {
        console.log(e)//这里是undefined
        console.log(e.target.value)
        if (e.target.value){
            getData({searchStr: e.target.value})
        }
    }
    const inputChangeDebounce = debounce(inputChange.bind(this),800)
    return (
        <View>
                    <Input onInput={inputChangeDebounce} type="text" value={searchStr} placeholder="输入搜索的关键词" confirm-type="search" />
        </View>
    )
}`

**其他信息**
[如截图等其他信息可以贴在这里]

**Comment by yesmeck** [comment]
试试用 useMemo 把 inputChangeDebounce 包起来。

**Reply by KouSum** [reply]
`    const inputChangeDebounce = useMemo( ()=>{
        return debounce(inputChange,800)
    },[1])`

是这样吗？但是还是报错app.js? [sm]:6136 TypeError: Cannot read property 'target' of undefined

**Reply by wuyine** [reply]
这里是 debounce 方法写错了，arguments 不应该在 setTimeout 中取，应该是return 的那个方法取。
